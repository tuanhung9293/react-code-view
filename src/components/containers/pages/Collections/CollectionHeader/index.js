import React from 'react'
import {connect} from 'react-redux'
import {objectToUrlParams} from 'services//transformers'
import {Columns, Column} from 'components/ui/grid'
import PrimaryButton from 'components/ui/buttons/PrimaryButton'
import styles from './CollectionHeader.module.css'

const CollectionHeader = ({searchParams}) => {
  return (
    <div className={styles.header}>
      <Columns>
        <Column span={8} breakPoints={[{maxWidth: 576, span: 12}]}>
          <div className={styles.title}>NASA COLLECTION</div>
        </Column>
        <Column span={4} breakPoints={[{maxWidth: 576, span: 12}]}>
          <div className={styles.addButton}>
            <PrimaryButton to={`/nasa-search${objectToUrlParams(searchParams)}`}>
              Add new item
            </PrimaryButton>
          </div>
        </Column>
      </Columns>
    </div>
  )
}

export default connect(
  ({medias}) => ({searchParams: medias.params})
)(CollectionHeader)
