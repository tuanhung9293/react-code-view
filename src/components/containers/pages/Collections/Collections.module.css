.loadMore {
    padding-top: 25px;
    padding-bottom: 40px;
    text-align: center;
}