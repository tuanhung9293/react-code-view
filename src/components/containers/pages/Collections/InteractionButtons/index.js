import React from 'react'
import {connect} from 'react-redux'
import LikeButton from 'components/ui/buttons/LikeButton'
import DeleteButton from 'components/ui/buttons/DeleteButton'
import EditButton from 'components/ui/buttons/EditButton'
import {likeCollection, removeCollection} from 'state/actions/collectionsAction'
import styles from './InteractionButtons.module.css'

const InteractionButtons = ({itemData, likedCollection, deleteCollection}) => {
  return (
    <div className={styles.base}>
      <LikeButton color={`${itemData.liked ? 'red' : 'white'}`} onClick={() => likedCollection(itemData.nasaId)} />
      <DeleteButton onClick={() => deleteCollection(itemData.nasaId)} />
      <EditButton to={`/collection/edit/${itemData.nasaId}`} />
    </div>
  )
}

export default connect(
  null,
  (dispatch) => ({
    likedCollection: (collectionId) => dispatch(likeCollection(collectionId)),
    deleteCollection: (collectionId) => dispatch(removeCollection(collectionId))
  })
)(InteractionButtons)
