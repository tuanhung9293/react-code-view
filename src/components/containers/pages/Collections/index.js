import React from 'react'
import {connect} from 'react-redux'
import {PAGINATION_UNIT} from 'constants/defaultValues'
import DocumentTitle from 'components/html/DocumentTitle'
import {Columns, Column} from 'components/ui/grid'
import PrimaryButton from 'components/ui/buttons/PrimaryButton'
import SortSelector from 'components/ui/forms/SortSelector'
import MediaTypeCheckbox from 'components/ui/forms/MediaTypeCheckbox'
import CollectionHeader from './CollectionHeader'
import InteractionButtons from './InteractionButtons'
import MediasGalaxy from '../../shared/MediasGalaxy'
import {sortNasaCollections} from 'state/actions/collectionsAction'
import styles from './Collections.module.css'

class Collections extends React.Component {
  state = {
    sortValue: 'dateEdited,1',
    image: true,
    video: true,
    audio: true,
    favorite: false,
    page: 1
  }

  componentDidMount = () => {
    this.props.sortCollections(this.state.sortValue)
  }

  sortHandle = (e) => {
    e.preventDefault()
    this.setState({sortValue: e.target.value})
    this.props.sortCollections(e.target.value)
  }

  onImageCheck = (e) => {
    this.setState({image: e.target.checked}, this.paginatedCollection)
  }

  onVideoCheck = (e) => {
    this.setState({video: e.target.checked}, this.paginatedCollection)
  }

  onAudioCheck = (e) => {
    this.setState({audio: e.target.checked}, this.paginatedCollection)
  }

  onFavoriteCheck = (e) => {
    this.setState({favorite: e.target.checked}, this.paginatedCollection)
  }

  onLoadMore = () => {
    this.setState({page: this.state.page + 1}, this.paginatedCollection)
  }

  filterCollections = () => {
    const {image, video, audio, favorite} = this.state
    const {collectionsData} = this.props

    const filledCollections = collectionsData.filter((item) => {
      const evalFavorite = favorite ? (favorite && item.liked) : true
      return ((image && item.mediaType === 'image') ||
      (video && item.mediaType === 'video') ||
      (audio && item.mediaType === 'audio')) && evalFavorite
    })

    return filledCollections
  }

  paginatedCollection = () => {
    const {page} = this.state
    return this.filterCollections().slice(0, page * PAGINATION_UNIT)
  }

  render() {
    const {sortValue, video, image, audio, favorite, page} = this.state
    const {collectionsData} = this.props
    return (
      <React.Fragment>
        <DocumentTitle title="Collections" />
        <div>
          <CollectionHeader />
          <Columns>
            <Column span={8} breakPoints={[{maxWidth: 576, span: 12}]}>
              <div className={styles.typeCheckbox}>
                <MediaTypeCheckbox
                  label="Filter by"
                  video={video} image={image} audio={audio} favorite={favorite}
                  onImageCheck={this.onImageCheck}
                  onVideoCheck={this.onVideoCheck}
                  onAudioCheck={this.onAudioCheck}
                  onFavoriteCheck={this.onFavoriteCheck}
                />
              </div>
            </Column>
            <Column span={4} breakPoints={[{maxWidth: 576, span: 12}]}>
              <SortSelector label="Sort by" value={sortValue} onChange={this.sortHandle} />
            </Column>
          </Columns>
        </div>
        <MediasGalaxy
          galaxyData={this.paginatedCollection()}
          source="collections"
          actionComponent={InteractionButtons}
          total={collectionsData.length}
        />
        {this.filterCollections().length > page * PAGINATION_UNIT &&
          <div className={styles.loadMore}>
            <PrimaryButton onClick={this.onLoadMore}>Load more</PrimaryButton>
          </div>
        }
      </React.Fragment>
    )
  }
}

export default connect(
  ({collections}) => ({collectionsData: collections.data}),
  (dispatch) => ({
    sortCollections: (value) => dispatch(sortNasaCollections(value))
  })
)(Collections)
