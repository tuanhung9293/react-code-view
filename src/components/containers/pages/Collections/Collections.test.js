import React from 'react'
import {shallow} from 'enzyme'
import {Provider} from 'react-redux'
import Collections from './'

import configureStore from 'redux-mock-store'
const mockStore = configureStore()

let store = mockStore({
  collections: {
    data: [{}, {}, {}]
  }
})
store.dispatch = () => {}

describe('renders', () => {
  it('renders Collections', () => {
    expect(shallow(
      <Provider store={store}>
        <Collections />
      </Provider>
    )).toMatchSnapshot()
  })
})

