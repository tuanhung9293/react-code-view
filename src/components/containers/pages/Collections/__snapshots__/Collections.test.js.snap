// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders Collections 1`] = `<Connect(Collections) />`;
